src/ApiResourceExampleDocument.js

```diff
@@ -34,7 +34,14 @@ import elementStyles from './styles/Document.js';
  */
 export class ApiResourceExampleDocument extends AmfHelperMixin(LitElement) {
   get styles() {
-    return elementStyles;
+    return [
+      css`
+        .customMaxHeight {
+          max-height: ${unsafeCSS(this.maxHeight)};
```

Comment:
Please, use CSS variables instead of JS properties for styling consistency.

Reply:
Hi Pawel, thanks for your feedback, I'm try to follow your approach and follow related [documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties), but I'm block when I try set property from javascript beacause I don't have access to element of the DOM until that the Render method is invocated. 

I was try othe approach,  it was,  send directly to the tag the atrribute, something like that, `<div style="minHeight: ${this.minHeight}>` but I believe is a bit dirty

Please, Pawel can you pass me an article or example of the best practice in this kind of components and it styling code for customize css properties before of mount the component?

Reply:
Hi @alexpmule 
As I mentioned in the comment below the right approach would be to use CSS variables instead. You don't need access to the element to set a CSS variable. This can be done at any time. The CSS variables are propagated through the shadow tree.
You can take a look at this article to get familiar with the topic: https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties

Reply:
Also, 
| I don't have access to element of the DOM until that the Render method is invocated.
you do have access to the DOM. YOu may not have access to the element you are trying to render but the variables can be set anywhere in the DOM (above the target element) or better yet in a CSS file.

Reply:
I tried something like that
```javascript
 return [
      css`
        .renderer.customMaxHeight {
          max-height: var(--api-resource-example-document-max-height, ${unsafeCSS(this.maxHeight)});
        }
      `,
      elementStyles
    ];
```
How do you see my approach?

Reply:
Take a look at this example:
https://github.com/api-components/amf-components/blob/main/demo/demo.css#L32

These are variables for the demo page for the new components. They are declared in a regular CSS file. 
Eventually they are used in the demo page as a regular css: https://github.com/api-components/amf-components/blob/main/demo/api-documentation.html#L12

As you can see here, I don't need direct access to the element to set the variables. Full CSS rules apply here.